codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -57,15 +58,30 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.remoteFlowState(new DefaultFlowState(stream, 0));
+                // If the stream state is not open then the stream is not yet eligible for flow controlled frames and
+                // only requires the ReducedFlowState. Otherwise the full amount of memory is required.
+                stream.remoteFlowState(stream.state() == IDLE ?
```

Comment:
> It's reasonable to expect that any stream in idle will eventually go to open

Maybe...not sure if that is true....doesn't have to be the case if they are just using these streams to create priority groups of other streams.

> With "longer periods of time" being milliseconds to a few seconds.

Again...maybe not sure.  I think the point is we don't really know.  I think we should operate under the assumption that we don't allocate resources for streams unless we actually have to.  If this is causing a problem then we can re-evaluate.